hooks/useLessonUnlocked.ts

```diff
@@ -0,0 +1,54 @@
+import { useUser, usePathData } from 'hooks'
+import { chapters } from 'content'
+
+function removeAfterHyphen(str: string): string {
+  return str.split('-')[0]
+}
+
+function getIndex(arr: string[], searchStr: string): number {
+  return arr.indexOf(searchStr)
+}
+
+export const useLessonUnlocked = () => {
+  let { lang, pageId, chapterId, lessonId } = usePathData()
+  const userInfo = useUser()
+
+  // if someone vists the website for the first time
+  if (userInfo.user === undefined) {
+    return `/${lang}`
+  } else {
+    const userLesson = userInfo.user.progress.lesson
+    const userChapter = userInfo.user.progress.chapter
+    const lessonArray = chapters[chapterId].metadata.lessons
+    const reDirectURL = `/${lang}/${pageId}/${userChapter}/${userLesson}`
+
+    if (userChapter.split('-')[1] < chapterId.split('-')[1]) {
+      return reDirectURL
+    } else {
+      if (lessonId.includes('intro')) {
+        return true
+      } else {
+        if (lessonId.includes('outro')) {
```

Comment:
Why are we returning `true` if an intro or outro lesson is hit? If I understand correctly that means the user will _not_ be redirected while they should. If I haven't unlocked chapter-2 intro-1 I should also be redirected away right?

Reply:
The user will be redirected if they directly go to chapter-2 intro-1 without unlocking chapter-2 because the first `if` statement checks whether the particular chapter has been unlocked.

I have updated the logic. All the 3 different arrays (intros, lessons, and outros) are combined into a single array, and then a comparison is made based on the index of the saved userProgress and the lesson that the user is trying to access. Before I was checking for intro or outro page because they were not part of the array.